civet/compilers/coffeescript.py

```diff
@@ -75,11 +75,17 @@ def compile(self, src_path, dst_path):
         map_path = os.path.join(dst_dir, map_filename)
 
         if os.path.exists(map_path):
-            map_data = None
             with open(map_path) as f:
-                map_data = json.load(f)
-                map_data['sourceRoot'] = ''
-                map_data['sources'] = [os.path.basename(src_path)]
-
-            with open(map_path, 'w') as f:
-                json.dump(map_data, f)
+                try:
+                    map_data = json.load(f)
+                except ValueError as err:
+                    f.seek(0)
+                    print("Warning: could not read valid sourcemap JSON from "
+                          "{}. Exception is:\n{}\n\nContents are:\n{}"
+                          .format(map_path, err, f.read()))
+                    return
+                else:
+                    map_data['sourceRoot'] = ''
+                    map_data['sources'] = [os.path.basename(src_path)]
+                    with open(map_path, 'w') as f:
+                        json.dump(map_data, f)
```

Comment:
Is it a good idea to open two file handles to the same file, one in read mode and one in write mode? I wasn't able to find anything on the internet about this, but it worries me slightly (is there a potential to self deadlock?).

Since you have a `return` statement in the the `except` case, you could just close the original file handle and open a new one, similar to the old code.